components/common/hooks/useCanSelfHostedSignup.ts

```diff
@@ -1,18 +1,67 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {useState, useEffect} from 'react';
+import {useMemo, useState, useEffect} from 'react';
 
 import {Client4} from 'mattermost-redux/client';
 
 import useLoadStripe from './useLoadStripe';
 
-export default function useCanSelfHostedSignup() {
-    const [canReachPortal, setCanReachPortal] = useState(false);
-    const canReachStripe = (useLoadStripe().current);
+interface CWSSignupAvailability {
+    cwsContacted: boolean;
+    cwsServiceOn: boolean;
+}
+
+const cwsAvailable: CWSSignupAvailability = {
+    cwsContacted: true,
+    cwsServiceOn: true,
+};
+const cwsAvailableEmptyState: CWSSignupAvailability = {
+    cwsContacted: false,
+    cwsServiceOn: false,
+};
+
+type SignupAvailability = CWSSignupAvailability & {
+    stripeAvailable: boolean;
+    ok: boolean;
+}
+
+export default function useCanSelfHostedSignup(): SignupAvailability {
+    const [cwsAvailability, setCwsAvailability] = useState(cwsAvailableEmptyState);
+    const stripeAvailable = Boolean(useLoadStripe().current);
     useEffect(() => {
-        Client4.getAvailabilitySelfHostedSignup().then(() => setCanReachPortal(true));
+        Client4.getAvailabilitySelfHostedSignup().
+            then(() => {
+                setCwsAvailability(cwsAvailable);
+            }).
+            catch((err) => {
+                let errorValue = {...cwsAvailableEmptyState};
+                switch (err.status_code) {
+                case 503: {
+                    errorValue = {
+                        cwsServiceOn: false,
+                        cwsContacted: true,
+                    };
+                    break;
+                }
+
+                // defined as switch because in the future we will add a separate
+                // screening pending reason
+                default: {
+                    errorValue = {...cwsAvailableEmptyState};
+                }
+                }
+                setCwsAvailability(errorValue);
+            });
     }, []);
 
-    return canReachPortal && canReachStripe;
+    // not implemented
+
+    return useMemo(() => {
+        return {
+            ...cwsAvailability,
+            stripeAvailable,
+            ok: stripeAvailable && cwsAvailability.cwsContacted && cwsAvailability.cwsServiceOn,
```

Comment:
What would be the case if `cwsContacted == false`? It seems that it would always execute the code above where all code paths seem to set it to `true`

Reply:
It would happen for timed out requests (CWS is hard down) and for non 503 responses. with the other proposed changes in the other related MM-49772 PRs, if the implementation doesn't send the query param indicating support for the new request format, it will receive a 501.